File: python_csv/scrap-csv.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Dec  4 22:14:43 2023

@author: antony
"""

import requests
from bs4 import BeautifulSoup

# Scrape the CSV example from the Wikipedia page
url = "https://en.wikipedia.org/wiki/Comma-separated_values"
response = requests.get(url)
soup = BeautifulSoup(response.content, "html.parser")
csv_data = soup.find_all("pre")

# Save the scraped CSV data to a file in the "python_csv" folder
with open("cars.csv", "w") as file:
    file.write(csv_data[10].text)

File: r_csv/scrap-csv.R
# Load necessary libraries
library(tidyverse)
library(rvest)

# Load URL of the Wikipedia page
url <- "https://en.wikipedia.org/wiki/Comma-separated_values"

# Read the Wiki HTML content and parse it to 'page'
page <- read_html(url)

# Extract the CSV data, Convert XML nodeset to raw text & Parse to 'csv_data'
csv_data <- page %>%
  html_nodes(xpath = '//*[@id="mw-content-text"]/div[1]/pre[1]/text()') %>%
  html_text()

# Convert CSV to data frame
df <- read.csv(text = csv_data)

# Export CSV file
write_csv(df, 'cars.csv')
